consensus/src/quorum_store/direct_mempool_quorum_store.rs

```diff
@@ -80,16 +80,20 @@ impl DirectMempoolQuorumStore {
         callback: oneshot::Sender<Result<ConsensusResponse>>,
     ) {
         let get_batch_start_time = Instant::now();
-        let (txns, result) = match payload_filter {
-            PayloadFilter::DirectMempool(exclude_txns) => {
-                match self.pull_internal(max_txns, max_bytes, exclude_txns).await {
-                    Err(e) => {
-                        error!("GetBatch failed {:?}", e);
-                        (vec![], counters::REQUEST_FAIL_LABEL)
-                    }
-                    Ok(txns) => (txns, counters::REQUEST_SUCCESS_LABEL),
-                }
+        let exclude_txns = match payload_filter {
+            PayloadFilter::DirectMempool(exclude_txns) => exclude_txns,
+            PayloadFilter::InQuorumStore(_) => {
+                unreachable!("Unknown payload_filter: {}", payload_filter)
```

Comment:
is it really unreachable?

Reply:
Yes, I verified. We need to handle the special transition case. Need some real engineering advice here:)

Reply:
Seems not to be a problem since at a new epoch there is nothing to exclude.